biogeophys/EDSurfaceAlbedoMod.F90

```diff
@@ -288,9 +296,26 @@ subroutine PatchNormanRadiation (currentPatch, &
          do ft = 1,numpft
             currentPatch%canopy_mask(L,ft) = 0
             do  iv = 1, currentPatch%nrad(L,ft)
+               total_lai_sai(L,ft,iv) = currentPatch%elai_profile(L,ft,iv)+currentPatch%esai_profile(L\
+,ft,iv)
                if (currentPatch%canopy_area_profile(L,ft,iv) > 0._r8)then
                   currentPatch%canopy_mask(L,ft) = 1
-                  !I think 'present' is only used here...
+                  ! layer level reflectance qualities
+                  do ib = 1,hlm_numSWb !vis, nir
+                    if(total_lai_sai(L,ft,iv).gt.0._r8)then
+                      frac_lai = currentPatch%elai_profile(L,ft,iv)/total_lai_sai(L,ft,iv)
+                      frac_sai = 1.0_r8 - frac_lai
+                      f_abs(L,ft,iv,ib) = 1.0_r8 - (frac_lai*(rhol(ft,ib) + taul(ft,ib))+&
```

Comment:
Is this initialized when total_lai_sai<=0? 
If uninitialized, could propogate nans, even in cases where it is multiplied by zero.

Reply:
Good catch @rgknox.  That might explain some of the slightly odd behaviour I'm currently chasing...

Reply:
I've taken that if statement out now...